src/interface/title_sequences.cpp

```diff
@@ -0,0 +1,607 @@
+    {
+        command.command = TITLE_SCRIPT_LOADMM;
+    }
+    else if (String::Equals(command_text, "loadrct1", true))
+    {
+        command.command = TITLE_SCRIPT_LOADRCT1;
+        command.saveIndex = (uint8)json_integer_value(json_object_get(json, "saveIndex"));
+    }
+    return command;
+}
+
+// Parses a title sequence from json
+//		json: The json to parse
+//		title_sequence: A pointer to the sequence to update
+void title_sequence_from_json(json_t* json, title_sequence* sequence)
+{
+    sequence->num_commands = (uint16) json_array_size(json);
+    sequence->commands = Memory::AllocateArray<title_command>(sequence->num_commands);
+
+    for (int i = 0; i < sequence->num_commands; i++)
+        sequence->commands[i] = title_sequence_command_from_json(json_array_get(json, i), sequence);
+}
+
+void title_sequence_open(const char *path, const char *customName)
+{
+    utf8 scriptPath[MAX_PATH], titlePath[MAX_PATH];
+    file_info fileInfo;
+    int fileEnumHandle, preset;
+
+    // Check for the script file
+    String::Set(scriptPath, MAX_PATH, path);
+    safe_strcat_path(scriptPath, "script.json", MAX_PATH);
+
+    if (!platform_file_exists(scriptPath))
+    {
+        // No script file, title sequence is invalid
+        log_error("Title sequence script file doesn't exist: %s\n", scriptPath);
+        return;
+    }
+
+    // Check if the preset is already loaded
+    // No need to read the first two presets as they're hardcoded in
+    for (preset = 0; preset < gConfigTitleSequences.num_presets; preset++)
+        if (String::Equals(path, gConfigTitleSequences.presets[preset].name, true))
+            return;
+
+    // Otherwise allocate one
+    if (preset == gConfigTitleSequences.num_presets)
+    {
+        gConfigTitleSequences.num_presets++;
+        gConfigTitleSequences.presets = Memory::ReallocateArray(gConfigTitleSequences.presets, (size_t) gConfigTitleSequences.num_presets);
+
+        if (customName == NULL)
+        {
+            StringBuilder* nameBuffer = new StringBuilder();
+            nameBuffer->Append(path);
+            // Get folder name
+            // First find the last index of the separator
+            size_t folderNameEnd = String::LastIndexOf(nameBuffer->GetBuffer(), *PATH_SEPARATOR);
+            // Remove everything after that
+            StringBuilder* nameBuffer2 = nameBuffer->SubString(0, folderNameEnd);
+            // Find the separator before it
+            size_t folderNameStart = String::LastIndexOf(nameBuffer2->GetBuffer(), *PATH_SEPARATOR);
+            // Remove everything befoe that
+            StringBuilder* nameBuffer3 = nameBuffer2->SubString(folderNameStart + 1, nameBuffer2->GetLength());
+            
+            Memory::CopyArray<char>(gConfigTitleSequences.presets[preset].name, nameBuffer3->GetBuffer(), TITLE_SEQUENCE_NAME_SIZE);
+            gConfigTitleSequences.presets[preset].path[0] = '\0';
+            delete nameBuffer, nameBuffer2, nameBuffer3;
```

Comment:
That's not intended use of comma operator. It does not do what you expect it to.